libs/domain/cart/add/src/add.component.ts

```diff
@@ -79,20 +79,35 @@ export class CartAddComponent extends ProductMixin(
   };
 
   protected $hasStock = computed(() => {
+    const product = this.$product();
+
+    if (product?.discontinued && !product?.availability?.quantity) return false;
```

Comment:
What about wrap `product?.discontinued && !product?.availability?.quantity` into some getter to avoid duplication (isDiscontinued)? Maybe even on product mixin level

Reply:
good one, it became a bit messy indeed. I've refactored the min/max logic and simplified it all.